### PR TITLE
Features/string camel case test

### PR DESCRIPTION
**Pull request by agamble33**
Added one method test, testing out camelCase

### CODE REVIEW

**Review by agamble33 — COMMENTED** [review]
Please edit spaces between variables, and reveiw expected results for tests, and what you want the outcome to be.

File: src/test/javascript/stringExtensionSpec.js [hunk 1]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+            var testString = "Wario";
+
+            var expectedResult = "wario";
+
+            var actualResult = testString.toCamelCase();
+
+            expect(actualResult).toBe(expectedResult);
+        });
+
+            it('will stay the same', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:134]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 2]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 3]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var testString = "darkPit";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 4]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var expectedResult = "darkPit";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 5]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var testString = "pokemon trainer";
+
+                var expectedResult = "pokemonTrainer";
+
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+        it('will minimize first letter of word', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 6]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var testString = "RosalinaLuma";
+
+                var expectedResult = "rosalinaLuma";
+
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('will keep first letter of word minimized, capitalize first letter of second word and condense', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:118]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 7]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var expectedResult = "mrCooper";
+
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+
+            it('will minimize capital letter of first word and condense', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:110]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 8]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+
+                var expectedResult = "captainFalcon";
+
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will minimize first and second letters of first word and keep capitalize on second word and condense', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:101]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 9]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
 
                 expect(result).toBe("helloThisIsASentence.");
             });
+
+            it('will minimize first and second letters of first word and keep capital on second word and condense', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:92]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 10]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
+
+                var expectedResult = "Donkey Kong";
+
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will trim whitespace correctly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 11]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
 
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
-                var result = "                       l     shdasdk;lg;al j          ".trim();
 
-                expect(result).toBe("l     shdasdk;lg;al j");
+
+
+
+            it('will trim whitespace correctly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 12]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+        it('will minimize first letter of word', function () {
+
+            var testString = "Wario";
+
+            var expectedResult = "wario";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:130]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 13]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+        it('will minimize first letter of word', function () {
+
+            var testString = "Wario";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:130]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 14]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var expectedResult = "pokemonTrainer";
+
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+        it('will minimize first letter of word', function () {
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:130]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 15]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('will keep first letter of word minimized, capitalize first letter of second word and condense', function () {
+
+                var testString = "pokemon trainer";
+
+                var expectedResult = "pokemonTrainer";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 16]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('will keep first letter of word minimized, capitalize first letter of second word and condense', function () {
+
+                var testString = "pokemon trainer";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 17]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+
+            it('will minimize capital letter of first word and condense', function () {
+                var testString = "RosalinaLuma";
+
+                var expectedResult = "rosalinaLuma";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 18]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+
+            it('will minimize capital letter of first word and condense', function () {
+                var testString = "RosalinaLuma";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 19]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var expectedResult = "rosalinaLuma";
+
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('will keep first letter of word minimized, capitalize first letter of second word and condense', function () {
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 20]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will minimize first and second letters of first word and keep capitalize on second word and condense', function () {
+
+                var testString = "MrCooper";
+
+                var expectedResult = "mrCooper";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 21]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will minimize first and second letters of first word and keep capitalize on second word and condense', function () {
+
+                var testString = "MrCooper";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 22]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
...
+                var expectedResult = "captainFalcon";
+
+                var actualResult = testString.toCamelCase();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will minimize first and second letters of first word and keep capitalize on second word and condense', function () {
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 23]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
 
                 expect(result).toBe("helloThisIsASentence.");
             });
+
+            it('will minimize first and second letters of first word and keep capital on second word and condense', function () {
+
+                var testString = "captainFalcon";
+
+                var expectedResult = "captainFalcon";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 24]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
 
                 expect(result).toBe("helloThisIsASentence.");
             });
+
+            it('will minimize first and second letters of first word and keep capital on second word and condense', function () {
+
+                var testString = "captainFalcon";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 25]
```diff
@@ -31,6 +53,75 @@ define(['bytepushers'], function(BytePushers) {
 
                 expect(result).toBe("helloThisIsASentence.");
             });
+
+            it('will minimize first and second letters of first word and keep capital on second word and condense', function () {
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 26]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will trim whitespace correctly', function () {
+
+                var testString = " Duck Hunt";
+
+                var expectedResult = "Duck Hunt";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:53]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 27]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will trim whitespace correctly', function () {
+
+                var testString = " Duck Hunt";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:50]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 28]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
+                var expectedResult = "Donkey Kong";
+
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+
+            });
+
+            it('will trim whitespace correctly', function () {
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:45]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 29]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
-                expect(result).toBe("l     shdasdk;lg;al j");
+
+
+
+            it('will trim whitespace correctly', function () {
+
+                var testString = " Donkey Kong ";
+
+                var expectedResult = "Donkey Kong";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:25]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 30]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
-                var result = "                       l     shdasdk;lg;al j          ".trim();
 
-                expect(result).toBe("l     shdasdk;lg;al j");
+
+
+
+            it('will trim whitespace correctly', function () {
+
+                var testString = " Donkey Kong ";
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:16]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 31]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
-                var result = "                       l     shdasdk;lg;al j          ".trim();
 
-                expect(result).toBe("l     shdasdk;lg;al j");
+
+
+
+            it('will trim whitespace correctly', function () {
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 32]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
         var testStr = "Hello this is a random string";
 
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
-                var result = "                       l     shdasdk;lg;al j          ".trim();
 
-                expect(result).toBe("l     shdasdk;lg;al j");
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:16]: Remove empty line.

File: src/test/javascript/stringExtensionSpec.js [hunk 33]
```diff
@@ -4,10 +4,32 @@ define(['bytepushers'], function(BytePushers) {
...
+
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:20]: Remove empty line.

**Review by tonte-pouncil — CHANGES_REQUESTED** [review]

File: src/test/javascript/stringExtensionSpec.js (hunk 32 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:16]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 33 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:20]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 11 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: auto reformat puts space between function ()

File: src/test/javascript/stringExtensionSpec.js (hunk 31 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 30 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:16]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 29 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:25]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 28 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:45]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 27 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:50]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 26 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:53]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 9 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:92]: Reformat option on webstorm puts the space back between function ()

File: src/test/javascript/stringExtensionSpec.js (hunk 17 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 1 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:134]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 2 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Empty line removed.

File: src/test/javascript/stringExtensionSpec.js (hunk 4 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Empty line removed.

File: src/test/javascript/stringExtensionSpec.js (hunk 5 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 3 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 6 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:118]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 7 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:110]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 8 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:101]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 12 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:130]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 13 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:130]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 14 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:130]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 24 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 25 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 23 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 22 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 21 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 19 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 20 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 18 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 16 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

File: src/test/javascript/stringExtensionSpec.js (hunk 15 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed empty line.

**Pull request merged**